lib/events.js

```diff
@@ -425,9 +425,9 @@ EventEmitter.prototype.listeners = function listeners(type) {
     if (!evlistener)
       ret = [];
     else if (typeof evlistener === 'function')
-      ret = [evlistener];
+      ret = [evlistener.listener || evlistener];
     else
-      ret = arrayClone(evlistener, evlistener.length);
```

Comment:
But I think the only difference between is that `arrayClone` supports a copy with a specified length, 2nd argument and the assignment from `.listener` of elements.

The function `upwrapListeners` seems to be only used inside to `EE.prototype.listeners`, so could we declare this function inside the prototype method to be `unwrap`? Both the functions defined at bottom looks like those pure collection ones.